src/artm/regularizer/topic_segmentation_ptdw.cc

```diff
@@ -16,107 +17,272 @@ namespace regularizer {
 
 void TopicSegmentationPtdwAgent::Apply(int item_index, int inner_iter,
                                        ::artm::utility::LocalPhiMatrix<float>* ptdw) const {
-  const int local_token_size = ptdw->no_rows();
-  const int topic_size = ptdw->no_columns();
+  int local_token_size = ptdw->no_rows();
+  int num_topics = ptdw->no_columns();
   std::vector<float> background_probability(local_token_size, 0.0f);
-
-  // if background topics are given, count probability for each word to be background
+  //// if background topics are given, count probability for each word to be background
+  std::vector<bool> is_background_topic = core::is_member(args_.topic_name(), config_.background_topic_names());
   if (config_.background_topic_names().size()) {
-    std::vector<bool> is_background_topic = core::is_member(args_.topic_name(), config_.background_topic_names());
+    //std::vector<bool> is_background_topic = core::is_member(args_.topic_name(), config_.background_topic_names());
     for (int i = 0; i < local_token_size; ++i) {
       const float* local_ptdw_ptr = &(*ptdw)(i, 0);  // NOLINT
-      for (int k = 0; k < topic_size; ++k) {
+      for (int k = 0; k < num_topics; ++k) {
         if (is_background_topic[k]) {
           background_probability[i] += local_ptdw_ptr[k];
         }
       }
     }
   }
+  //for (int i = 0; i < local_token_size; ++i) {
```

Comment:
I see, that this code is dirty, but try not to pull request temp comments, you can store them in a local branch (thx git:))